File: src/main/java/com/skuniv/bigdata/domain/dto/open_api/CharterWithRentItemDto.java
package com.skuniv.bigdata.domain.dto.open_api;

import lombok.*;

import javax.xml.bind.annotation.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@XmlRootElement(name = "item")
@XmlAccessorType(XmlAccessType.FIELD)
public class CharterWithRentItemDto {
    @XmlElement(name = "보증금액")
    private String guaranteePrice;

    @XmlElement(name = "월세금액")
    private String monthlyPrice;

    @XmlElement(name = "건축년도")
    private int constructYear;

    @XmlElement(name = "년")
    private int year;

    @XmlElement(name = "법정동")
    private String dong;

    @XmlElements({
            @XmlElement(name = "아파트"),
            @XmlElement(name = "연립다세대"),
            @XmlElement(name = "단지")
    })
    private String name;

    @XmlElement(name = "월")
    private int monthly;

    @XmlElement(name = "일")
    private String days;

    @XmlElement(name = "전용면적")
    private Double area;

    @XmlElement(name = "지번")
    private String buildingNum;

    @XmlElement(name = "지역코드")
    private String regionCode;

    @XmlElement(name = "층")
    private int floor;
}

File: src/main/java/com/skuniv/bigdata/domain/dto/open_api/CharterWithRentItemDto.kt
package com.skuniv.bigdata.domain.dto.open_api

import lombok.Builder
import lombok.Data
import javax.xml.bind.annotation.*

@XmlRootElement(name = "item")
@XmlAccessorType(XmlAccessType.FIELD)
data class CharterWithRentItemDto(
        @field:XmlElement(name = "보증금액")
        var guaranteePrice: String? = null,

        @field:XmlElement(name = "월세금액")
        var monthlyPrice: String? = null,

        @field:XmlElement(name = "건축년도")
        var constructYear: Int = 0,

        @field:XmlElement(name = "년")
        var year: Int = 0,

        @field:XmlElement(name = "법정동")
        var dong: String? = null,

        @field:XmlElement(name = "아파트")
        var name: String? = null,

        @field:XmlElement(name = "월")
        var monthly: Int = 0,

        @field:XmlElement(name = "일")
        var days: String? = null,

        @field:XmlElement(name = "전용면적")
        var area: Double? = null,

        @field:XmlElement(name = "지번")
        var buildingNum: String? = null,

        @field:XmlElement(name = "지역코드")
        var regionCode: String? = null,

        @field:XmlElement(name = "층")
        var floor: Int = 0
)